synapse/rest/client/v2_alpha/account.py

```diff
@@ -875,6 +884,45 @@ async def on_POST(self, request):
         return 200, {"id_server_unbind_result": id_server_unbind_result}
 
 
+def assert_valid_next_link(hs: "HomeServer", next_link: str):
+    """
+    Raises a SynapseError if a given next_link value is invalid
+
+    next_link is valid if the scheme is http(s) and the next_link.domain_whitelist config
+    option is either empty or contains a domain that matches the one in the given next_link
+
+    Args:
+        hs: The homeserver object
+        next_link: The next_link value given by the client
+
+    Raises:
+        SynapseError: If the next_link is invalid
+    """
+    valid = True
+
+    # Parse the contents of the URL
+    next_link_parsed = urlparse(next_link)
+
+    # Scheme must be http(s)
+    if next_link_parsed.scheme not in ["http", "https"]:
+        valid = False
```

Comment:
I wonder whether this is too restrictive. Someone may wish to redirect to some otherprotocol (`ipfs://`, `dat://`, etc.). Perhaps this should still just disallow `file:///` again.

Reply:
I think it's better to just disallow `file:///` but ymmv.

Reply:
Yeah, I don't want to break someone's Riot hosted inside a firefox extension (`moz-extension://` or something). I'll change it back.